x/stake/client/rest/tx.go

```diff
@@ -320,11 +309,22 @@ func delegationsRequestHandlerFn(cdc *wire.Codec, kb keys.Keybase, cliCtx contex
 
 		output, err := wire.MarshalJSONIndent(cdc, results[:])
 		if err != nil {
-			w.WriteHeader(http.StatusInternalServerError)
-			w.Write([]byte(err.Error()))
+			utils.WriteErrorResponse(&w, http.StatusInternalServerError, err.Error())
 			return
 		}
 
 		w.Write(output)
 	}
 }
+
+func enrichContextWithGas(txCtx authcliCtx.TxContext, cliCtx context.CLIContext, name, password string, msg sdk.Msg) (authcliCtx.TxContext, int, error) {
```

Comment:
Can we deduplicate this function? This really shouldn't need to be in the module-specific REST files anyways I think.